aten/src/ATen/core/ivalue.cpp

```diff
@@ -463,6 +463,52 @@ std::ostream& IValue::repr(
   }
 }
 
+bool operator<(const ivalue::Tuple& lhs, const ivalue::Tuple& rhs) {
+  const auto& lhs_vals = lhs.elements();
+  const auto& rhs_vals = rhs.elements();
+
+  // Comparing tuples with different length is undefined.
+  TORCH_INTERNAL_ASSERT(lhs_vals.size() == rhs_vals.size());
+
+  const size_t n = lhs_vals.size();
+  for(size_t i = 0; i < n; ++i) {
+    if(lhs_vals[i] < rhs_vals[i]) {
+      return true;
+    }
+    if(lhs_vals[i] > rhs_vals[i]) {
+      return false;
+    }
+  }
+
+  // Reaching here means rhs and lhs are equal.
+  return false;
+}
+
+bool operator<(const IValue& a, const IValue& b) {
+  // Must be same type to be compared.
+  TORCH_INTERNAL_ASSERT(a.tag == b.tag);
+  switch (a.tag) {
+    case IValue::Tag::Tensor:
+       return a.toTensor().lt(b.toTensor()).is_nonzero();
+    case IValue::Tag::Double:
+       return a.toDouble() < b.toDouble();
+    case IValue::Tag::Int:
+       return a.toInt() < b.toInt();
+    case IValue::Tag::Bool:
+       return a.toBool() == false && b.toBool() == true;
+    case IValue::Tag::String:
+       return a.toString()->string() < b.toString()->string();
+    case IValue::Tag::Tuple:
+       return *a.toTuple() < *b.toTuple();
+    default:
+      AT_ERROR("IValue of type: ", a.tagKind(), " is not comparable");
+  }
+}
+
+bool operator>(const IValue& a, const IValue& b) {
+  return !(a == b || a < b);
```

Comment:
I wonder if there is an opportunity for performance improvement by implementing this explicitly instead of in terms of `==` and `<`.

But this proposal comes at the cost of increasing code size. I don't have strong feelings either way, just something to think about.

Reply:
Good point. Done.